pkg/cache/sizedlru/sizedlru.go

```diff
@@ -0,0 +1,120 @@
+// Package sizedlru provides a generic, size-limited, LRU (Least Recently Used) cache with optional
+// metrics collection and reporting. It wraps the golang-lru/v2 caching library, adding support for custom
+// metrics tracking cache hits, misses, evictions, and other cache operations.
+//
+// This package supports configuring key aspects of cache behavior, including maximum cache size,
+// and custom metrics collection.
+package sizedlru
+
+import (
+	"fmt"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/cache"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+)
+
+// collector is an interface that extends cache.BaseMetricsCollector
+// and adds methods for recording cache hits, misses, and evictions.
+type collector interface {
+	cache.BaseMetricsCollector
+
+	RecordEviction(cacheName string)
+}
+
+// Cache is a generic LRU-sized cache that stores key-value pairs with a maximum size limit.
+// It wraps the lru.Cache library and adds support for custom metrics collection.
+type Cache[T any] struct {
+	cache *lru.Cache[string, T]
+
+	cacheName string
+	capacity  int
+	metrics   collector
+}
+
+// Option defines a functional option for configuring the Cache.
+type Option[T any] func(*Cache[T])
+
+// WithMetricsCollector is a functional option to set a custom metrics collector.
+// It sets the metrics field of the Cache.
+func WithMetricsCollector[T any](collector collector) Option[T] {
+	return func(lc *Cache[T]) { lc.metrics = collector }
+}
+
+// WithCapacity is a functional option to set the maximum capacity of the cache.
+// If the capacity is not set, the default value (512MB) is used.
+func WithCapacity[T any](capacity int) Option[T] {
+	return func(lc *Cache[T]) { lc.capacity = capacity }
+}
```

Comment:
Can you document the units expected here?

Reply:
Yep, great idea.

Reply:
Whoops, I got the cache implementations mixed up. This capacity refers to the number of items, not total size.